### PR TITLE
Added EXTRACT(<time unit> FROM <date/datetime/time col) as partition pruning function

### PR DESCRIPTION
**Pull request by mjonss**
### What is changed, added or deleted? (Required)

See https://github.com/pingcap/tidb/issues/54210 / https://github.com/pingcap/tidb/pull/54209

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):
https://github.com/pingcap/tidb/issues/54210 / https://github.com/pingcap/tidb/pull/54209

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: partition-pruning.md [hunk 1]
```diff
@@ -240,6 +240,7 @@ If the `fn` function is monotonous, for any `x` and `y`, if `x > y`, then `fn(x)
 
 * [`UNIX_TIMESTAMP()`](/functions-and-operators/date-and-time-functions.md)
 * [`TO_DAYS()`](/functions-and-operators/date-and-time-functions.md)
+* [`EXTRACT(<time unit> FROM <DATETIME/DATE/TIME column>)`](/functions-and-operators/date-and-time-functions.md); For DATE and DATETIME columns YEAR, and YEAR_MONTH time units are considered monotonous functions, and for TIME columns HOUR, HOUR_MINUTE, HOUR_SECOND and HOUR_MICROSECOND are considered monotonous. Also note that WEEK is not supported as time unit in EXTRACT for partition pruning.
```
Comment (hfxsd) [inline partition-pruning.md:243]:
```suggestion
* [`EXTRACT(<time unit> FROM <DATETIME/DATE/TIME column>)`](/functions-and-operators/date-and-time-functions.md). For `DATE` and `DATETIME` columns, `YEAR` and `YEAR_MONTH` time units are considered monotonous functions. For the `TIME` column, `HOUR`, `HOUR_MINUTE`, `HOUR_SECOND` and `HOUR_MICROSECOND` are considered monotonous. Note that `WEEK` is not supported as time unit in `EXTRACT` for partition pruning.
```

File: partitioned-table.md [hunk 2]
```diff
@@ -1310,6 +1310,7 @@ Currently, partition pruning does not work with `LIKE` conditions.
 
     * [`UNIX_TIMESTAMP()`](/functions-and-operators/date-and-time-functions.md)
     * [`TO_DAYS()`](/functions-and-operators/date-and-time-functions.md)
+    * [`EXTRACT(<time unit> FROM <DATETIME/DATE/TIME column>)`](/functions-and-operators/date-and-time-functions.md); For DATE and DATETIME columns YEAR, and YEAR_MONTH time units are considered monotonous functions, and for TIME columns HOUR, HOUR_MINUTE, HOUR_SECOND and HOUR_MICROSECOND are considered monotonous. Also note that WEEK is not supported as time unit in EXTRACT for partition pruning.
```
Comment (hfxsd) [inline partitioned-table.md:1313]:
```suggestion
    * [`EXTRACT(<time unit> FROM <DATETIME/DATE/TIME column>)`](/functions-and-operators/date-and-time-functions.md). For `DATE` and `DATETIME` columns, `YEAR` and `YEAR_MONTH` time units are considered monotonous functions. For the `TIME` column, `HOUR`, `HOUR_MINUTE`, `HOUR_SECOND` and `HOUR_MICROSECOND` are considered monotonous. Note that `WEEK` is not supported as time unit in `EXTRACT` for partition pruning.
```

**Review by Frank945946 — APPROVED** [review]

**Review by Defined2014 — APPROVED** [review]

File: partition-pruning.md [hunk 3]
```diff
@@ -240,6 +240,7 @@ If the `fn` function is monotonous, for any `x` and `y`, if `x > y`, then `fn(x)
 
 * [`UNIX_TIMESTAMP()`](/functions-and-operators/date-and-time-functions.md)
 * [`TO_DAYS()`](/functions-and-operators/date-and-time-functions.md)
+* [`EXTRACT(<time unit> FROM <DATETIME/DATE/TIME column>)`](/functions-and-operators/date-and-time-functions.md). For `DATE` and `DATETIME` columns, `YEAR` and `YEAR_MONTH` time units are considered monotonous functions. For the `TIME` column, `HOUR`, `HOUR_MINUTE`, `HOUR_SECOND` and `HOUR_MICROSECOND` are considered monotonous. Note that `WEEK` is not supported as time unit in `EXTRACT` for partition pruning.
```
Comment (lilin90) [inline partition-pruning.md:243]: Keep wording consistent?
```suggestion
* [`EXTRACT(<time unit> FROM <DATETIME/DATE/TIME column>)`](/functions-and-operators/date-and-time-functions.md). For `DATE` and `DATETIME` columns, `YEAR` and `YEAR_MONTH` time units are considered monotonous functions. For the `TIME` column, `HOUR`, `HOUR_MINUTE`, `HOUR_SECOND` and `HOUR_MICROSECOND` are considered monotonous functions. Note that `WEEK` is not supported as time unit in `EXTRACT` for partition pruning.
```

File: partitioned-table.md [hunk 4]
```diff
@@ -1310,6 +1310,7 @@ Currently, partition pruning does not work with `LIKE` conditions.
 
     * [`UNIX_TIMESTAMP()`](/functions-and-operators/date-and-time-functions.md)
     * [`TO_DAYS()`](/functions-and-operators/date-and-time-functions.md)
+    * [`EXTRACT(<time unit> FROM <DATETIME/DATE/TIME column>)`](/functions-and-operators/date-and-time-functions.md). For `DATE` and `DATETIME` columns, `YEAR` and `YEAR_MONTH` time units are considered monotonous functions. For the `TIME` column, `HOUR`, `HOUR_MINUTE`, `HOUR_SECOND` and `HOUR_MICROSECOND` are considered monotonous. Note that `WEEK` is not supported as time unit in `EXTRACT` for partition pruning.
```
Comment (lilin90) [inline partitioned-table.md:1313]:
```suggestion
    * [`EXTRACT(<time unit> FROM <DATETIME/DATE/TIME column>)`](/functions-and-operators/date-and-time-functions.md). For `DATE` and `DATETIME` columns, `YEAR` and `YEAR_MONTH` time units are considered monotonous functions. For the `TIME` column, `HOUR`, `HOUR_MINUTE`, `HOUR_SECOND` and `HOUR_MICROSECOND` are considered monotonous functions. Note that `WEEK` is not supported as time unit in `EXTRACT` for partition pruning.
```

**Review by lilin90 — APPROVED** [review]
/lgtm

**Review by hfxsd — APPROVED** [review]

**Pull request merged**